kolibri_instant_schools_plugin/auth/api.py

```diff
@@ -1,15 +1,221 @@
 from django.conf import settings
-
-from kolibri.auth.models import Facility
+from django.db import transaction
+from django.utils.translation import ugettext as _
+from rest_framework import filters, permissions, status, viewsets, serializers
+from rest_framework.response import Response
+from kolibri.auth.models import Facility, FacilityUser
 from kolibri.auth.api import SignUpViewSet
 
-from .hashing import get_hash_value
+from .mapping import get_usernames, create_new_username
+from ..models import PasswordResetToken, PhoneToUsernameMapping
+from ..smpp.utils import send_password_reset_link, SMPPConnectionError
+
+from kolibri.auth.serializers import FacilityUserSerializer
+
+
+class PhoneNumberSignupFacilityUserSerializer(FacilityUserSerializer):
+
+    def validate_username(self, value):
+        if FacilityUser.objects.filter(username__iexact=value).exists():
+            raise serializers.ValidationError(_('An account already exists for this phone number. To add a new profile ' +
+                                                'under this account, you must first login. If you have forgotten your ' +
+                                                'password, you can reset it using the link on the login page.'))
```

Comment:
We'll probably want to trim this down to contain only the essential information: _An account already exists for this phone number_.

(https://material.io/guidelines/style/writing.html)